domains/grid/components/GridFloatingMenu.vue

```diff
@@ -0,0 +1,193 @@
+<script setup lang="ts">
+import { tv } from 'tailwind-variants'
+
+type Emits = {
+  (event: 'on-reset'): void
+  (event: 'on-save'): void
+  (event: 'on-add-grid'): void
+}
+
+const emits = defineEmits<Emits>()
+const { isEditingGrid, hasUnsavedGrid, isSavingGrid } =
+  storeToRefs(useGridStore())
+const { isConnected, isMobile } = storeToRefs(useAppStore())
+const { formatMessage } = useIntl()
+const { showModal } = useModal()
+
+const handleToggleGridEditMode = async () => {
+  isEditingGrid.value = !isEditingGrid.value
+}
+
+const handleAddContent = () => {
+  showModal({
+    template: 'AddGridWidget',
+    data: {
+      type: undefined, // when no type we display selection screen
```

Comment:
Does it make sense to create a new enum value?
Even though there aren't many instances that use `type: undefined` I think it's something that still can be useful even if it's used only locally and never uploaded as part of the config. No need to explain each time this value.

<img width="514" alt="Screenshot 2024-10-29 at 13 02 26" src="https://github.com/user-attachments/assets/72d02382-6163-407f-bd44-ccde201337c7">

Reply:
`undefined` is kind of defining here 😄

Reply:
Enum for `undefined` sounds weird, I will remove this as it will be better to replace with proper type.

Reply:
The suggestion is not an enum for undefined but rather an enum that would match "AddGridWidget". Throwing in an idea.